### ISSUE TITLE
How to access internal summary in Callback [question]

### ISSUE THREAD

**Issue by galatolofederico** [hill-a/stable-baselines#818]
Is there a way to access the fields logged in Tensorboard in a callback? The only value that seems accessible via `self.locals["self"]` is `episode_reward`.

**Comment by Solliet** [comment]
Every variable that is part of the 'train' method will be available once I get around to completing the documentation for #787 and I really lack time at the moment. If you really want to, you can clone the repo and pull the changes from that branch.

**Comment by galatolofederico** [comment]
Hey I have checked out #787 but as far as i can tell a lot of variables still remain inaccessible. It wouldn't be nice if all the training variables (for example in `A2C` loss, entropy, value_loss, etc...) somehow can be accessible from the callbacks? (so one can write stopping criterias using those values for example)

**Comment by araffin** [comment]
>so one can write stopping criterias using those values for example

I would suggest you to derive a custom class from the A2C class if you want to access those (as done in [https://github.com/araffin/learning-to-drive-in-5-minutes](https://github.com/araffin/learning-to-drive-in-5-minutes)).

**Comment by galatolofederico** [comment]
Sure thanks! If you think that it can be useful I can make a PR. Nothing too fancy just a `get_stats()` to all the algorithms.

**Comment by araffin** [comment]
>Nothing too fancy just a get_stats() to all the algorithms.

Can you elaborate a bit more?
Not sure if it is needed, as most users won't use it.

**Comment by galatolofederico** [comment]
I was thinking to save in a dictionary `self.training_values` all the training information (that in most algorithms are computed in the `_train_step()` function) and add to `BaseRLModel` a `get_training_values()` function to get them. Maybe it is better if i firstly work on the custom class solution to understand if this solution is really general for all the algorithms.
